Feature/ens

nishil-edx opened the pull request.

Commit:
.

Commit:
ens updated

Commit:
.

ex-rohan-edx commented on README.md:
```diff
@@ -1,49 +1 @@
-## License
-
-MIT
+This branch mainly targeted on ENS interactions
```
Revert back to old READ ME file

ex-rohan-edx commented on src/classes/EdexaClient.ts:
```diff
@@ -144,33 +144,50 @@ export class EdexaClient {
+        const provider = new ethers.providers.JsonRpcProvider(RPC_URL)
+        const reverseName = `${input.slice(2)}.addr.reverse`;
+        const node = ethers.utils.namehash(reverseName);
+        const resolverContract_ = new ethers.Contract('0xEF1db68FaDfdD398886EE73Cbc34918Be51Ae304', publicResolver.abi, provider);
```
Move static contract address into a constants file

ex-rohan-edx commented on src/classes/EdexaClient.ts:
```diff
@@ -144,33 +144,50 @@ export class EdexaClient {
-            throw new Error(`ENS Not Registered for ${input}`)
-          else return detailsObject.resolver
+        const node = ethers.utils.namehash(input);
+        const resolverContract_ = new ethers.Contract('0xEF1db68FaDfdD398886EE73Cbc34918Be51Ae304', publicResolver.abi, provider);
```
Move static contract address into a constants file

ex-rohan-edx commented on src/classes/EdexaClient.ts:
```diff
@@ -144,33 +144,50 @@ export class EdexaClient {
+  async resolveName(input: string) {                  //name string input
+    try {
+      // Check if the input is a valid Ethereum address
+      if (input.endsWith(".edx") && input.split(".")[0].length >= 4) {
```
Move static string checking var to a constants file

ex-rohan-edx commented on testENSresolve.ts:
```diff
@@ -0,0 +1,23 @@
+
+
+
+resolveAddress("0x44aDbAcC4B0baD81dBc6D24b9512da3806c5aCCD")
```
Add more test cases to check the functionality for example add negative test cases too

ex-rohan-edx commented on tsconfig.json:
```diff
@@ -18,6 +18,6 @@
     "declaration": true
   },
-  "include": ["./src/**/*"],
+  "include": ["./src/**/*", "testENSresolve.ts"],
```
Reset back to original file

nishil-edx committed:
resolver address moved to constants

nishil-edx committed:
revert readme

nishil-edx committed:
tsconfig reverted

nishil-edx committed:
test cases added

nishil-edx committed:
moved checking strings to constants

nishil-edx commented:
updated

On Wed, Apr 24, 2024 at 11:29 AM Rohan Jain ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/devEdexa/edexa.js/pull/9#discussion_r1577307281>:
>
> > -```
> -
> -### NPM publish
> -
> -Before running the publish script, make sure you have updated the version properly.
> -
> -Note that the `prepublishOnly` script will be automatically called while publishing. It will check lint, clean `dist/lib` folders, and build fresh distribution files before executing `npm publish`.
> -
> -```bash
> -npm publish
> -```
> -
> -## License
> -
> -MIT
> +This branch mainly targeted on ENS interactions
>
> Revert back to old READ ME file
> ------------------------------
>
> In src/classes/EdexaClient.ts
> <https://github.com/devEdexa/edexa.js/pull/9#discussion_r1577308810>:
>
> >      try {
>        // Check if the input is a valid Ethereum address
>        if (ethers.utils.isAddress(input)) {
> -        return input
> +
> +        const provider = new ethers.providers.JsonRpcProvider(RPC_URL)
> +        const reverseName = `${input.slice(2)}.addr.reverse`;
> +        const node = ethers.utils.namehash(reverseName);
> +        const resolverContract_ = new ethers.Contract('0xEF1db68FaDfdD398886EE73Cbc34918Be51Ae304', publicResolver.abi, provider);
>
> Move static contract address into a constants file
> ------------------------------
>
> In src/classes/EdexaClient.ts
> <https://github.com/devEdexa/edexa.js/pull/9#discussion_r1577309192>:
>
> > -          ensAbi,
> -          provider,
> -        )
> -
> -        // Resolve the ENS domain to an Ethereum address
> -        const detailsObject = await ens.getDomainInfo(input)
> -
> -        if (detailsObject) {
> -          if (
> -            detailsObject.resolver ==
> -            '0x0000000000000000000000000000000000000000'
> -          )
> -            throw new Error(`ENS Not Registered for ${input}`)
> -          else return detailsObject.resolver
> +        const node = ethers.utils.namehash(input);
> +        const resolverContract_ = new ethers.Contract('0xEF1db68FaDfdD398886EE73Cbc34918Be51Ae304', publicResolver.abi, provider);
>
> Move static contract address into a constants file
> ------------------------------
>
> In src/classes/EdexaClient.ts
> <https://github.com/devEdexa/edexa.js/pull/9#discussion_r1577309550>:
>
> >        } else {
> +        return "input is not an address";
> +
> +      }
> +    } catch (error) {
> +      throw new Error(`Error: ${error.message}`)
> +    }
> +  }
> +  async resolveName(input: string) {                  //name string input
> +    try {
> +      // Check if the input is a valid Ethereum address
> +      if (input.endsWith(".edx") && input.split(".")[0].length >= 4) {
>
> Move static string checking var to a constants file
> ------------------------------
>
> In testENSresolve.ts
> <https://github.com/devEdexa/edexa.js/pull/9#discussion_r1577310494>:
>
> > +
> +    let edexaclient = new EdexaClient();
> +    let ens = await edexaclient.resolveAddr(input);
> +    console.log("ens name of",input, " is:",ens)
> +
> +    }
> +async function resolveName(input : string) {
> +
> +    let edexaclient = new EdexaClient();
> +    let add = await edexaclient.resolveName(input);
> +    console.log("address of",input, " is:",add)
> +}
> +
> +
> +
> +resolveAddress("0x44aDbAcC4B0baD81dBc6D24b9512da3806c5aCCD")
>
> Add more test cases to check the functionality for example add negative
> test cases too
> ------------------------------
>
> In tsconfig.json
> <https://github.com/devEdexa/edexa.js/pull/9#discussion_r1577310804>:
>
> > @@ -18,6 +18,6 @@
>      "esModuleInterop": true,
>      "declaration": true
>    },
> -  "include": ["./src/**/*"],
> +  "include": ["./src/**/*", "testENSresolve.ts"],
>
> Reset back to original file
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/devEdexa/edexa.js/pull/9#pullrequestreview-2018908614>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHKYY3R24DUZGDLQICEO5MLY65C3RAVCNFSM6AAAAABGOPR2K6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDAMJYHEYDQNRRGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

ex-rohan-edx reviewed (changes requested):
**Function Comments:**
     - Add comments to each function describing:
     - The function's purpose.
     - The types of arguments it accepts.
     - What the function returns.

**Test Case Improvement:**
   - Specify the expected results in your test cases to clarify what the test is verifying.
   - Use descriptive names for your test cases to indicate the aspect of the function being tested.
